Lib/defconQt/glyphView.py

```diff
@@ -1380,59 +1676,122 @@ def __init__(self, glyph, parent=None):
         self.addBackground()
         self.addBlues()
         self.addHorizontalMetrics()
-        self.addOtherLayersOutlines()
-        self.addOutlines()
-        self.addComponents()
-        self.addAnchors()
-        self.addPoints()
 
-    def currentLayerBox(self):
-        return self._currentLayerBox
+        for layer in layerSet:
+            if self._name not in layer:
+                self._listenToLayer(layer)
+            else:
+                self._listenToGlyph(layer)
+        self._listenToLayerSet()
+
+        self.changeCurrentLayer(self._layer)
+
+
+    @property
+    def _glyph(self):
+        # instead of a getter we could change the glyph each time
+        # self._layer is changed. However, it might be that the glyph
+        # is not ready when the layer is set
```

Comment:
You first create a layer then you can create the glyph. The layer must be present before the glyph.